src/Aspire.Dashboard/Model/DashboardClient.cs

```diff
@@ -374,6 +374,38 @@ await foreach (var response in call.ResponseStream.ReadAllAsync(cancellationToke
         }
     }
 
+    public async Task<ResourceCommandResponseViewModel> ExecuteResourceCommandAsync(string resourceName, string resourceType, CommandViewModel command, CancellationToken cancellationToken)
+    {
+        EnsureInitialized();
+
+        var request = new ResourceCommandRequest()
+        {
+            CommandType = command.CommandType,
+            Parameter = command.Parameter,
+            ResourceName = resourceName,
+            ResourceType = resourceType
+        };
+
+        try
+        {
+            var response = await _client!.ExecuteResourceCommandAsync(request, cancellationToken: cancellationToken);
+
+            return response.ToViewModel();
+        }
+        catch (RpcException ex)
+        {
+            _logger.LogError(ex, "Error executing command \"{CommandType}\" on resource \"{ResourceName}\": {StatusCode}", command.CommandType, resourceName, ex.StatusCode);
+
+            var errorMessage = ex.StatusCode == StatusCode.Unimplemented ? "Command not implemented" : "Unknown error. See logs for details";
```

Comment:
Get text from resources

Reply:
After chatting about this offline, we're going to leave this hard-coded for the moment. Other error messages coming across the wire won't be localized either (at least for now) and rather than address the potential problems here (bringing loc into DashboardClient or changing the viewmodel to handle what was already just special-casing of the not implemented exception) we can address it later if it becomes a problem and perhaps those issues will be different/handled/non-existent by then.